### PR TITLE
CSF-tools: Allow type checking in story title

### PR DESCRIPTION
**Pull request by honzahruby**
## What I did

V7 disallows using expressions and variables for story titles. This PR allows the title to conform to a certain (nested) structure using `satisfies` operator.

## How to test

Run storybook on this branch and use `satisfies` in the story title definition

```ts
enum ComponentGroups {
    General = 'General'
}
type ComponentTitle<Group extends ComponentGroups> = `${Group}/${string}`;

export default {
    title: 'General/Button' satisfies ComponentTitle<ComponentGroups.General>,
    component: Button,
};
```

## Checklist

- [x] Make sure your changes are tested (stories and/or unit, integration, or end-to-end tests)
- [x] Make sure to add/update documentation regarding your changes
- [x] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

- [ ] If this PR should be tested against many or all sandboxes,
      make sure to add the `ci:merged` or `ci:daily` GH label to it.
- [ ] Make sure this PR contains **one** of the labels below.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`

### CODE REVIEW

File: code/lib/csf-tools/src/CsfFile.ts [hunk 1]
```diff
@@ -168,6 +168,8 @@ export class CsfFile {
       ? findVarInitialization(value.name, this._ast.program)
       : value;
     if (t.isStringLiteral(node)) return node.value;
+    if (t.isTSSatisfiesExpression(node)) return node.expression.value;
+
```
Comment (kasperpeulen) [inline code/lib/csf-tools/src/CsfFile.ts:172]: It seems like we want here that satisfies expression only be able to contain a string then.

**Review by kasperpeulen — CHANGES_REQUESTED** [review]
Looks good in general, but we would like to remain specific about what a title can be, even if it is a satisfies expression.

File: code/lib/csf-tools/src/CsfFile.ts (hunk 1 above)
Comment (honzahruby) [reply code/lib/csf-tools/src/CsfFile.ts:172]: Absolutely, thank you for pointing that out 👍 
I just wanted to hotfix it for my personal use and then forgot about this condition while making it public

**Review by kasperpeulen — APPROVED** [review]
LGTM!

**Comment by honzahruby** [comment]
What can I do to get this merged?

**Pull request merged**